aviary/subsystems/propulsion/turboprop_model.py

```diff
@@ -286,37 +337,260 @@ def setup(self):
+            #   torque output, so "shp_out" is an input to that internal component
+            gearbox_input_list = list(
+                set(
+                    gearbox_input_dict[key]['prom_name']
+                    for key in gearbox_input_dict
+                    if '.' not in gearbox_input_dict[key]['prom_name']
+                    and '_out' not in gearbox_input_dict[key]['prom_name']
+                )
+            )
+            gearbox_inputs = []
+            gearbox_output_dict = gearbox_model.list_outputs(
+                return_format='dict', units=True, out_stream=None, all_procs=True
+            )
+            gearbox_output_list = list(
+                set(
+                    gearbox_output_dict[key]['prom_name']
+                    for key in gearbox_output_dict
+                    if '.' not in gearbox_output_dict[key]['prom_name']
+                )
+            )
+            gearbox_outputs = []
+
+        if self.options['propeller_model'] is None:
+            propeller_model_name = 'propeller_model'
+        else:
+            propeller_model_name = self.options['propeller_model'].name
+        propeller_model = self._get_subsystem(propeller_model_name)
+        propeller_input_dict = propeller_model.list_inputs(
+            return_format='dict', units=True, out_stream=None, all_procs=True
+        )
+        propeller_input_list = list(
+            set(
+                propeller_input_dict[key]['prom_name']
+                for key in propeller_input_dict
+                if '.' not in propeller_input_dict[key]['prom_name']
+            )
+        )
+        propeller_inputs = []
+        # always promote all propeller model outputs w/o aliasing except thrust
+        propeller_outputs = [
+            '*',
+            (Dynamic.Mission.THRUST, 'propeller_thrust'),
+            (Dynamic.Mission.THRUST_MAX, 'propeller_thrust_max'),
+        ]
+
+        #########################
+        # SHP MODEL CONNECTIONS #
+        #########################
+        # Everything not explicitly handled here gets promoted later on
+        # Thrust outputs are directly promoted with alias (this is a special case)
+        if Dynamic.Mission.THRUST in shp_output_list:
+            shp_outputs.append((Dynamic.Mission.THRUST, 'turboshaft_thrust'))
+            shp_output_list.remove(Dynamic.Mission.THRUST)
+
+        if Dynamic.Mission.THRUST_MAX in shp_output_list:
+            shp_outputs.append((Dynamic.Mission.THRUST_MAX, 'turboshaft_thrust_max'))
+            shp_output_list.remove(Dynamic.Mission.THRUST_MAX)
+
+        # Gearbox connections
+        if has_gearbox:
+            for var in shp_output_list.copy():
+                # Check for case: var is output from shp_model, connects to gearbox, then
+                #   gets updated by gearbox
+                # RPM has special handling, so skip it here
+                if (
+                    var in shp_output_list
```

Comment:
this might be unnecessary, since we are using `for var in shp_output_list`

Reply:
Yep I think you're right - removing